common/dllserver/thorplugin.cpp

```diff
@@ -379,6 +233,207 @@ extern bool getResourceFromFile(const char *filename, MemoryBuffer &data, const
 
 //-------------------------------------------------------------------------------------------------------------------
 
+class HelperDll : implements ILoadedDllEntry, public CInterface
+{
+    SharedObject so;
+    StringAttr name;
+    Linked<const IFileIO> dllFile;
+    Owned<IMemoryMappedFile> mappedDll;
+    bool logLoad;
+public:
+    IMPLEMENT_IINTERFACE;
+    HelperDll(const char *_name, const IFileIO *dllFile);
+    ~HelperDll();
+
+//interface ILoadedDllEntry
+    virtual HINSTANCE getInstance() const;
+    virtual void * getEntry(const char * name) const;
+    virtual bool IsShared();
+    virtual const char * queryVersion() const;
+    virtual const char * queryName() const;
+    virtual const byte * getResource(unsigned id) const;
+    virtual bool getResource(size32_t & len, const void * & data, const char * type, unsigned id, bool trace) const;
+
+    bool load(bool isGlobal, bool raiseOnError);
+    bool loadCurrentExecutable();
+    bool loadResources();
+
+    virtual void logLoaded();
+    virtual bool checkVersion(const char *expected);
+};
+
+class PluginDll : public HelperDll
+{
+    ECLPluginDefinitionBlockEx pb;
+
+public:
+    PluginDll(const char *_name, const IFileIO *_dllFile) : HelperDll(_name, _dllFile) {}
+
+    bool init(IPluginContextEx * pluginCtx);
+
+    virtual bool checkVersion(const char *expected);
+    virtual void logLoaded();
+};
+
+HelperDll::HelperDll(const char *_name, const IFileIO *_dllFile)
+: name(_name), dllFile(_dllFile)
+{
+    logLoad = false;
+}
+
+bool HelperDll::load(bool isGlobal, bool raiseOnError)
+{
+    if (!so.load(name, isGlobal, raiseOnError))
+        return false;
+    return true;
+}
+
+bool HelperDll::loadResources()
+{
```

Comment:
It is a counterpart to load() and locadCurrentExecutable() - neither of which check if it is already loaded.  It isn't a public function, so can't be called more than once.